Extensions/audio_plus.js

```diff
@@ -106,56 +126,54 @@ XKit.extensions.audio_plus = {
 	},
 
 	create_pop_out_controls: function() {
-		var audio_plus = XKit.extensions.audio_plus;
-
-		// Create play and pause buttons modeled after the built-in ones
-		var playPause = document.createElement("div");
-		playPause.classList.add("play-pause");
-
-		var icon = document.createElement("i");
-		icon.classList.add("icon");
-		icon.classList.add("icon_pause");
-		playPause.appendChild(icon);
-
-		var controls_undock = document.createElement("div");
-		controls_undock.id = "xkit-audio-plus-controls-undock";
-
-		var controls_undock_container = document.createElement("div");
-		controls_undock_container.id = "xkit-audio-plus-controls-undock-container";
-		controls_undock_container.appendChild(controls_undock);
-
-		var psuedo_post = document.createElement("div");
-		psuedo_post.classList.add("xkit-audio-plus-pseudo-post");
-
-		var controls = document.createElement("div");
-		controls.classList.add("xkit-audio-plus-controls");
-		// Spoof audio_player class to get the Play/Pause button styles
-		controls.classList.add("audio-player");
-
-		var progress = document.createElement("div");
-		progress.classList.add("progress");
-		
-		var audio_info = document.createElement("div");
-		audio_info.classList.add("audio-info");
+		const cl = {
+			controls: "'xkit-audio-plus-controls audio-player'",
+			progress: "'progress'",
+			playPause: "'play-pause'",
+			icon: "'icon icon_pause'",
+			audio_info: "'audio-info'",
+			track_name: "'track-name'",
+			track_artist: "'track-artist'",
+			audio_image: "'audio-image'",
+		};
 		
-		var track_name = document.createElement("div");
-		track_name.classList.add("track-name");
-		track_name.innerHTML = "Listen";
+		const id = {
+			controls_undock_container: "'xkit-audio-plus-controls-undock-container'",
+			controls_undock: "'xkit-audio-plus-controls-undock'",
+		};
 		
-		var track_artist = document.createElement("div");
-		track_artist.classList.add("track-artist");
-
-		var audio_image = document.createElement("div");
-		audio_image.classList.add("audio-image");
+		const controls_markup = `
+			<div class=${cl.controls}>
```

Comment:
would be nicer if you used `<div class="${cl.controls}">` etc rather than putting each of your strings in quotes. they're part of the HTML, not the classes, so putting them in the class variables is kinda confusing

Reply:
why are we interpolating these variables at all if they're constant? IMO this should be:

```

<div class="xkit-audio-plus-controls audio-player">
    <div class="progress"></div>
        <div class="play-pause">
           <i class="icon icon_pause"></i>
    </div>
    <div class="audio-info">
        <div class="track-name"></div>
        <div class="audio-image"></div>
    </div>
</div>
<div id="xkit-audio-plus-controls-undock-container">
    <div id="xkit-audio-plus-controls-undock"></div>
</div>
```

I'm also not super happy about the mix of class-based and id-based scoping. we should use one or the other—and there's basically no reason to use `id` in modern web design

Reply:
i agree the interpolation is a bit redundant, but i don't understand why a mix of class and id is bad :? to clarify though, i basically followed the html that this extension had in 0.5.0